File: python/design-patterns/iterator/concrete/concrete_iterators.py
class ProductIterator(Iterator):
    def __init__(self, products):
        self.products = products
        self.index = 0
    def has_next(self):
        return self.index < len(self.products)
    def next(self):
        p = self.products[self.index]
        self.index += 1
        return p

File: java/design-patterns/iterator/concrete/ConcreteIterators.java
class ProductIterator implements Iterator<String> {
    private List<String> list;
    private int index = 0;
    public ProductIterator(List<String> list) { this.list = list; }
    public boolean hasNext() { return index < list.size(); }
    public String next() { return list.get(index++); }
}